videos/threeplay_api.py

```diff
@@ -0,0 +1,129 @@
+"""3play api requests"""
+import logging
+from urllib.request import Request, urlopen
+
+import boto3
+import requests
+from django.conf import settings
+from requests import HTTPError
+
+from videos.constants import DESTINATION_YOUTUBE
+
+
+log = logging.getLogger(__name__)
+
+
+def threeplay_updated_media_file_request():
+    """3play data request to get files with 'updated' tag"""
+    payload = {"label": "updated", "api_key": settings.THREEPLAY_API_KEY}
+    url = "https://api.3playmedia.com/v3/files"
+    response = requests.get(url, payload)
+    if response:
+        return response.json()
+    else:
+        return {}
+
+
+def threeplay_remove_tags(threeplay_video_id):
+    """3play patch to remove tag from video file"""
+    payload = {"label": "", "api_key": settings.THREEPLAY_API_KEY}
+    url = f"https://api.3playmedia.com/v3/files/{threeplay_video_id}"
+    requests.patch(url, payload)
+
+
+def threeplay_transcript_api_request(youtube_id):
+    """3play data requst to get transcripts by youtube_id"""
+    payload = {
+        "media_file_reference_id": youtube_id,
+        "api_key": settings.THREEPLAY_API_KEY,
+    }
+    url = "https://api.3playmedia.com/v3/transcripts"
+    response = requests.get(url, payload)
+    if response:
+        return response.json()
+    else:
+        return {}
+
+
+def upload_to_s3(source_url, s3_destination):
+    """Upload transcripts to s3"""
+
+    request = Request(source_url)
+
+    try:
+        response = urlopen(request)
+    except HTTPError:
+        log.error(
+            "Could not open 3play transcript at %s",
+            source_url,
+        )
+        return False
+
+    s3 = boto3.resource("s3")
+    bucket_name = settings.AWS_STORAGE_BUCKET_NAME
+    bucket = s3.Bucket(bucket_name)
+
+    bucket.upload_fileobj(
+        Fileobj=response, Key=s3_destination, ExtraArgs={"ACL": "public-read"}
+    )
+
+    return True
+
+
+def update_transcripts_for_video(video):
+    """Download transcripts from 3play, upload them to s3 and update the Video file"""
+
+    youtube_video_file = video.videofiles.filter(
+        destination=DESTINATION_YOUTUBE
+    ).first()
+
+    if not (youtube_video_file and youtube_video_file.destination_id):
+        return
+    else:
+        youtube_id = youtube_video_file.destination_id
+
+    threeplay_transcript_json = threeplay_transcript_api_request(youtube_id)
+
+    if (
+        threeplay_transcript_json.get("data")
+        and len(threeplay_transcript_json.get("data")) > 0
+    ):
+        transcript_id = threeplay_transcript_json["data"][0].get("id")
+        media_file_id = threeplay_transcript_json["data"][0].get("media_file_id")
+
+        s3_base = (
+            f"{settings.S3_TRANSCRIPTS_PREFIX}/{video.website.short_id}/{youtube_id}/"
```

Comment:
I wonder if the base url prefix should be the same as for uploaded website files, so that the current publishing pipeline definition will automatically copy these from the OCW Studio S3 bucket to the preview/publish buckets?